library/schema_check.py

```diff
@@ -79,9 +79,9 @@ def check_file_paths(module, *args):
 
 def validate_grp_names(data):
     res_grps = data['resource_groups']
-    res_grp_vars = data['resource_group_vars']
+    res_grp_vars = data['resource_group_vars'] if 'resource_group_vars' in data.keys() else []
     res_grp_names = [ x['resource_group_name'] for x in res_grps ]
-    res_grp_vars = [ x['resource_group_name'] for x in res_grp_vars ]
+    res_grp_vars = [ x['resource_group_name'] for x in res_grp_vars ] if len(res_grp_vars) > 0 else []
```

Comment:
@herlo In the above scenario , i am using if else statement as a ternery operator ,
http://book.pythontips.com/en/latest/ternary_operators.html
res_grp_vars are defaulted to empty list , if not mentioned.
For readability it can be changed to as follows : 
```
if len(res_grp_vars) > 0:
    # essentially all dicts of res_grp_vars into a list
    res_grp_vars = [ x['resource_group_name'] for x in res_grp_vars ]
else:
    res_grp_vars = []
```